mapie/conformity_scores/utils.py

```diff
@@ -1,26 +1,197 @@
+    Check parameter ``conformity_score`` for regression task.
 
     Parameters
     ----------
-    y_pred_proba: NDArray of shape (n_samples, n_classes)
-        Model prediction.
+    conformity_score: BaseClassificationScore
+        Conformity score function.
+
+        By default, `None`.
+
+    sym: bool
+        Whether to use symmetric bounds.
+
+        By default, `True`.
+
+    Raises
+    ------
+    ValueError
+        If parameters are not valid.
+
+    Examples
+    --------
+    >>> from mapie.conformity_scores.utils import (
+    ...     check_regression_conformity_score
+    ... )
+    >>> try:
+    ...     check_regression_conformity_score(1)
+    ... except Exception as exception:
+    ...     print(exception)
+    ...
+    Invalid conformity_score argument.
+    Must be None or a ConformityScore instance.
+    """
+    if conformity_score is None:
+        return AbsoluteConformityScore(sym=sym)
+    elif isinstance(conformity_score, BaseRegressionScore):
+        return conformity_score
+    else:
+        raise ValueError(
+            "Invalid conformity_score argument.\n"
+            "Must be None or a ConformityScore instance."
+        )
 
-    y: NDArray of shape (n_samples)
-        Labels.
 
-    Returns
-    -------
-    NDArray of shape (n_samples, 1)
-        Position of the true label in the prediction.
+def _check_depreciated(
+    method: str
+) -> None:
     """
-    index = np.argsort(np.fliplr(np.argsort(y_pred_proba, axis=1)))
-    position = np.take_along_axis(index, y.reshape(-1, 1), axis=1)
+    Check if the chosen method is outdated.
+
+    Raises
+    ------
+    Warning
+        If method is ``"score"`` (not ``"lac"``) or
+        if method is ``"cumulated_score"`` (not ``"aps"``).
+    """
+    if method == "score":
+        warnings.warn(
+            "WARNING: Deprecated method. "
+            + "The method \"score\" is outdated. "
+            + "Prefer to use \"lac\" instead to keep "
+            + "the same behavior in the next release.",
+            DeprecationWarning
+        )
+    if method == "cumulated_score":
+        warnings.warn(
+            "WARNING: Deprecated method. "
+            + "The method \"cumulated_score\" is outdated. "
+            + "Prefer to use \"aps\" instead to keep "
+            + "the same behavior in the next release.",
+            DeprecationWarning
+        )
+
+
+def check_target(
+    conformity_score: BaseClassificationScore,
+    y: ArrayLike
+) -> None:
+    """
+    Check that if the type of target is binary,
+    (then the method have to be ``"lac"``), or multi-class.
+
+    Parameters
+    ----------
+    conformity_score: BaseClassificationScore
+        Conformity score function.
 
-    return position
+    y: NDArray of shape (n_samples,)
+        Training labels.
+
+    Raises
+    ------
+    ValueError
+        If type of target is binary and method is not ``"lac"``
+        or ``"score"`` or if type of target is not multi-class.
+    """
```

Comment:
Why adding the lambda ? you can write `score: APS` so the method is not instanciate and you do it only when you call it

Reply:
type-check problem

Reply:
Should be at the top of the file

Reply:
Done 👍